flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcDynamicTableFactory.java

```diff
@@ -196,6 +196,7 @@ private JdbcOptions getJdbcOptions(ReadableConfig readableConfig) {
 		final JdbcOptions.Builder builder = JdbcOptions.builder()
 			.setDBUrl(url)
 			.setTableName(readableConfig.get(TABLE_NAME))
+			.setConnectionCheckTimeoutSeconds((int) readableConfig.get(MAX_RETRY_TIMEOUT).toMillis())
```

Comment:
This is wrong. `toMillis()` returns milliseconds but the parameter requires seconds. Should use `toSeconds()` here.

Reply:
We should better to check the configured timeout is greater than 1 second. Otherwise, we will get a zero seconds which means never timeout for `Connection#isValid(timeout)`.